core/src/main/java/ai/timefold/solver/core/api/score/buildin/bendablelong/BendableLongScore.java

```diff
@@ -138,9 +131,9 @@ public int initScore() {
     }
 
     /**
-     * @return not null, array copy because this class is immutable
+     * @return array copy because this class is immutable
      */
-    public long[] hardScores() {
+    public long @NonNull [] hardScores() {
```

Comment:
For my edification, as I haven't seen this before... is this how you have to annotate array types? My brain would find `@NonNull long[]` much more natural.